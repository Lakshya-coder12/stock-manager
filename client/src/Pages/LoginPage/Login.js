import React, { Component } from "react";
import Button from "../../Components/Button/Button";
import PropTypes from "prop-types";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
    };
  }
  handleInputChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  handleRedirect = () => {
    this.props.history.push("/signup");
  };

  handleLogin = (e) => {
    e.preventDefault();
    const formData = {
      email: this.state.email,
      password: this.state.password,
    };
    this.props.login(formData, this.props.history);
  };
  render() {
    const { email, password } = this.state;
    return (
      <>
        <form className="form" onSubmit={this.handleLogin}>
          <div className="input-element">
            <label className="label">Email</label>
            <input
              required
              name="email"
              onChange={this.handleInputChange}
              value={email}
              className="input"
              type="email"
            />
          </div>
          <div className="input-element">
            <label className="label">Password</label>
            <input
              required
              name="password"
              className="input"
              onChange={this.handleInputChange}
              value={password}
              type="password"
            />
          </div>
          <div className="form-btn">
            <Button>Log In</Button>
          </div>
          <div onClick={this.handleRedirect} className="redirect">
            Don't have an account? Click here to create one
          </div>
        </form>
      </>
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
};

export default Login;
