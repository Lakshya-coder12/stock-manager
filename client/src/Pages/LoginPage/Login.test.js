import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Login from "./Login";

configure({ adapter: new Adapter() });

describe("Testing Login page", () => {
  it("matches snapshot", () => {
    const login = jest.fn();
    const wrapper = shallow(<Login login={login} />);
    expect(wrapper).toMatchSnapshot();
  });
});
