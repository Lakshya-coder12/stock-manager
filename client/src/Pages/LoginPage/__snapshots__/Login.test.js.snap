// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login page matches snapshot 1`] = `
<Fragment>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Email
      </label>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="input-element"
    >
      <label
        className="label"
      >
        Password
      </label>
      <input
        className="input"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-btn"
    >
      <Button>
        Log In
      </Button>
    </div>
    <div
      className="redirect"
      onClick={[Function]}
    >
      Don't have an account? Click here to create one
    </div>
  </form>
</Fragment>
`;
